api/spec/controllers/mno_enterprise/jpi/v1/app_instances_controller_spec.rb

```diff
@@ -6,85 +6,84 @@ module MnoEnterprise
-        allow(organization).to receive(:app_instances).and_return(app_instances)
-        # Only active instances
-        allow(app_instances).to receive(:active).and_return(app_instances)
-        # Updated since last tick
-        allow(app_instances).to receive(:where).and_return([app_instance])
-      end
+      let(:app_instance) { build(:app_instance, status: 'running' , under_free_trial: false) }
+
+      before { stub_api_v2(:get, '/app_instances', [app_instance], [:app, :owner], {filter: {owner_id: organization.id, status: MnoEnterprise::AppInstance::ACTIVE_STATUSES}}) }
 
       before { sign_in user }
-      let(:timestamp) { nil }
-      subject { get :index, organization_id: organization.id, timestamp: timestamp }
+      subject { get :index, organization_id: organization.id }
 
-      it_behaves_like "jpi v1 protected action"
+      it_behaves_like 'jpi v1 protected action'
 
       describe 'all' do
-        it { subject; expect(assigns(:app_instances)).to eq([app_instance]) }
-
-        it 'filter only active instances' do
-          expect(app_instances).to receive(:active)
+        it {
           subject
-        end
-      end
-
-      context 'with timestamp' do
-        let(:timestamp) { Time.current.to_i }
-
-        it 'filter updates since last request' do
-          expect(app_instances).to receive(:where).with("updated_at.gt" => Time.at(timestamp))
-          subject
-        end
+          # TODO: Test that the rendered json is the expected one
+          # expect(assigns(:app_instances)).to eq([app_instance])
+          assert_requested(:get, api_v2_url('/app_instances', [:app, :owner], {filter: {owner_id: organization.id, status: MnoEnterprise::AppInstance::ACTIVE_STATUSES}}))
+        }
       end
 
       context 'without access to the app_instance' do
         before { allow(ability).to receive(:can?).with(any_args).and_return(false) }
-        before { subject }
-        it { expect(assigns(:app_instances)).to be_empty }
+        it {
+          subject
+          expect(assigns(:app_instances)).to be_empty
+        }
       end
     end
 
     describe 'POST #create' do
-      let(:app) { build(:app, nid: 'my-app' ) }
-      let(:app_instance) { build(:app_instance, app: app, owner: organization ) }
+      before { stub_audit_events }
+      let(:app) { build(:app, nid: 'my-app') }
+      let(:app_instance) { build(:app_instance, app: app, owner: organization, owner_id: organization.id) }
       subject { post :create, organization_id: organization.id, nid: 'my-app' }
 
       before do
-        api_stub_for(post: "/organizations/#{organization.id}/app_instances", response: from_api(app_instance))
-        api_stub_for(get: "/organizations/#{organization.id}/app_instances", response: from_api([app_instance]))
+        stub_api_v2(:post, '/app_instances/provision', app_instance)
       end
-      it_behaves_like "jpi v1 protected action"
+      it_behaves_like 'jpi v1 protected action'
+      it {
```

Comment:
Sorry, Rubymine is displaying that with a warning, and I automatically replace the " by `